Makefile

```diff
@@ -174,41 +188,82 @@ redeploy:
-        kubectl delete -f deploy/kubedirector/deployment-prebuilt.yaml --now; \
-    fi
-	@echo
-	@echo \* Deleting custom resource definitions...
-	-kubectl delete -f deploy/kubedirector/kubedirector_v1alpha1_kubedirectorapp_crd.yaml --now
-	-kubectl delete -f deploy/kubedirector/kubedirector_v1alpha1_kubedirectorcluster_crd.yaml --now
-	-kubectl delete -f deploy/kubedirector/kubedirector_v1alpha1_kubedirectorconfig_crd.yaml --now
-	@echo
-	@echo \* Deleting service account...
-	-@if [[ -f deploy/kubedirector/rbac.yaml ]]; then \
-        echo kubectl delete -f deploy/kubedirector/rbac.yaml --now; \
-        kubectl delete -f deploy/kubedirector/rbac.yaml --now; \
-    else \
-        echo kubectl delete -f deploy/kubedirector/rbac-default.yaml --now; \
-        kubectl delete -f deploy/kubedirector/rbac-default.yaml --now; \
-    fi
+	@true; \
+        function delete_thing { \
+            if [[ "$$3" == "" ]]; then \
+                namespace_arg=""; \
+                kind=$$1; \
+                name=$$2; \
+            else \
+                namespace_arg=" -n $$1"; \
+                kind=$$2; \
+                name=$$3; \
+            fi; \
+            cmd="kubectl$$namespace_arg delete $$kind $$name --now"; \
+            msg=$$($$cmd 2>&1); \
+            if [[ "$$?" == "0" ]]; then \
+                echo $$cmd; \
+                if [[ "$$msg" != "" ]]; then \
+                    echo "$$msg"; \
+                fi; \
+            else \
+                if [[ ! "$$msg" =~ "Error from server (NotFound):" ]]; then \
+                    echo $$cmd; \
+                    echo "$$msg"; \
+                    exit 1; \
+                fi; \
+            fi; \
+        }; \
+        function delete_all_things { \
+            cmd="kubectl -n $$1 delete $$2 --all --now"; \
+            msg=$$($$cmd 2>&1); \
+            if [[ "$$?" == "0" ]]; then \
+                if [[ "$$msg" != "No resources found" ]]; then \
+                    echo $$cmd; \
+                    if [[ "$$msg" != "" ]]; then \
+                        echo "$$msg"; \
+                    fi; \
+                fi; \
+            else \
+                if [[ ! "$$msg" =~ "the server doesn't have a resource type" ]]; then \
+                    echo $$cmd; \
+                    echo "$$msg"; \
+                    exit 1; \
+                fi; \
+            fi; \
+        }; \
+        all_namespaces=`kubectl get ns --no-headers| awk '{print $$1}'`; \
+        echo \* Deleting any managed virtual clusters...; \
+        for ns in $$all_namespaces; do \
+            delete_all_things $$ns ${cluster_resource_name}; \
+        done; \
+        echo; \
+        echo \* Deleting any application types...; \
+        for ns in $$all_namespaces; do \
+            delete_all_things $$ns ${app_resource_name}; \
+        done; \
+        echo; \
+        echo \* Deleting any configs...; \
+        for ns in $$all_namespaces; do \
```

Comment:
Same comment as above. It's unnecessary to loop through the namespaces. You call delete all of these resources with a single command.

kubectl delete kubedirectorconfig --all --all-namespaces --now

This obviously also applies to all of the other delete loops.

Reply:
Tracking in issue #206.